posixlib/src/main/scala/scala/scalanative/runtime/SocketHelpers.scala

```diff
@@ -24,7 +23,7 @@ object SocketHelpers {
       var hints = alloc[addrinfo]
       var ret   = alloc[Ptr[addrinfo]]
 
-      string.memset(hints.asInstanceOf[Ptr[Byte]], 0, sizeof[addrinfo])
+      libc.memset(hints.rawptr, 0, sizeof[addrinfo])
```

Comment:
The change gets rid of a cast.  I spent more time that I admit trying to figure out which of the two uses of libc was relevant here, and in ratting through includes to figure it out. 

I believe that a special variant of libc and of memset() are being used. Withot
further info, I expected the libc variant (yes, that libc ;-) which one I leave as
an exercise for the Gentle Reader).
If I got it right, please consider `runtime.libc.memset`. I think that would give a 
better clue.

If I got it wrong, then I am clueless (could be! here and elsewhere) and could use
a better clue or more coffee.

Reply:
This is using the special runtime `nativelib` `libc.memset` which is needed so that the runtime doesn't depend on `clib`. It uses this one so it doesn't depend on `clib`.

The `SocketHelpers.scala` really belongs in the portion of `javalib` that depends on `posixlib`. It is in the `runtime` package but this needs to be a temporary home until a future PR. It should be private to Scala Native so people should not be using it.